vxsandbox/rlimiter.py

```diff
@@ -1,102 +1,78 @@
 # -*- test-case-name: vxsandbox.tests.test_sandbox_rlimiter -*-
 
-"""A helper for apply RLIMITS to a sandboxed process.
-
-NOTE::
-
-    This module is also used as a standalone Python program that is executed by
-    the sandbox machinery. It must never, ever import non-stdlib modules.
+"""
+A helper for applying RLIMITS to a sandboxed process.
 """
 
-import os
-import sys
-import json
-import signal
 import resource
 
 
 class SandboxRlimiter(object):
-    """This reads rlimits in from stdin, applies them and then execs a
-    new executable.
+    """
+    Spawn a process that applies rlimits and then execs another program.
 
     It's necessary because Twisted's spawnProcess has no equivalent of
     the `preexec_fn` argument to :class:`subprocess.POpen`.
 
     See http://twistedmatrix.com/trac/ticket/4159.
     """
-    def __init__(self, argv, env):
-        start = argv.index('--') + 1
-        self._executable = argv[start]
-        self._args = [self._executable] + argv[start + 1:]
-        self._env = env
-
-    def _apply_rlimits(self):
-        data = os.environ[self._SANDBOX_RLIMITS_]
-        rlimits = json.loads(data) if data.strip() else {}
-        for rlimit, (soft, hard) in rlimits.iteritems():
-            # Cap our rlimits to the maximum allowed.
-            rsoft, rhard = resource.getrlimit(int(rlimit))
-            soft = min(soft, rsoft)
-            hard = min(hard, rhard)
-            resource.setrlimit(int(rlimit), (soft, hard))
-
-    def _reset_signals(self):
-        # reset all signal handlers to their defaults
-        for i in range(1, signal.NSIG):
-            if signal.getsignal(i) == signal.SIG_IGN:
-                signal.signal(i, signal.SIG_DFL)
-
-    def _sanitize_fds(self):
-        # close everything except stdin, stdout and stderr
-        maxfds = resource.getrlimit(resource.RLIMIT_NOFILE)[1]
-        os.closerange(3, maxfds)
-
-    def execute(self):
-        self._apply_rlimits()
-        self._restore_child_env(os.environ)
-        self._sanitize_fds()
-        self._reset_signals()
-        os.execvpe(self._executable, self._args, self._env)
-
-    _SANDBOX_RLIMITS_ = "_SANDBOX_RLIMITS_"
 
-    @classmethod
-    def _override_child_env(cls, env, rlimits):
-        """Put RLIMIT config in the env."""
-
-        env[cls._SANDBOX_RLIMITS_] = json.dumps(rlimits)
+    ULIMIT_PARAMS = {
+        resource.RLIMIT_CORE: "c",
+        resource.RLIMIT_CPU: "t",
+        resource.RLIMIT_FSIZE: "f",
+        resource.RLIMIT_DATA: "d",
+        resource.RLIMIT_STACK: "s",
+        resource.RLIMIT_RSS: "m",
+        resource.RLIMIT_NOFILE: "n",
+        resource.RLIMIT_MEMLOCK: "l",
+        resource.RLIMIT_AS: "v",
+    }
+
+    # These ones can't always be set, so ignore them.
```

Comment:
Turns out this was actually a problem with value scaling. I'll fix that instead.